usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -1549,6 +1549,8 @@ arc_cksum_is_equal(arc_buf_hdr_t *hdr, zio_t *zio)
 
 		void *cbuf = zio_buf_alloc(HDR_GET_PSIZE(hdr));
 		csize = zio_compress_data(compress, zio->io_abd, cbuf, lsize);
+		abd_t *cdata = abd_get_from_buf(cbuf, HDR_GET_PSIZE(hdr));
+		abd_take_ownership_of_buf(cdata, B_TRUE);
```

Comment:
Is this effectively doing the same thing as:
```
abd_t *cdata = abd_alloc_linear(HDR_GET_PSIZE(hdr), B_TRUE)
zio_compress_data(... abd_to_buf(cdata) ...)
```
If so, let's simplify it to that.

Reply:
make sense. will do.